### DISCUSSION TITLE
Questions about GFM renewable energy system establishing.

### DISCUSSION THREAD

**Discussion by xiaoheimai95** [CURENT/andes#431] [Q&A] [answered]
Dear Dr. Cui and Jinning,
As we discussed before, I need to build a renewable energy system of Grid-Forming. I would like to know whether there is a GFM control module, or we just need to directly use REGF1 in the model library as a generator unit (as in the case I provided). In addition, I found that if all the traditional generating units were replaced with REGF1 units, the system would fail to initialize or fail to convergence. Hope to get your advice~!

**Comment by jinningwang** [comment, marked as answer]
Hello @xiaoheimai95,

Thanks for your interest.
As far as I know, there are no existing separate control modules for [REGCV1](https://docs.andes.app/en/latest/groupdoc/RenGen.html#regcv1), [REGCV2](https://docs.andes.app/en/latest/groupdoc/RenGen.html#regcv2), [REGF1](https://docs.andes.app/en/latest/groupdoc/RenGen.html#regf1), [REGF2](https://docs.andes.app/en/latest/groupdoc/RenGen.html#regf2), and [REGF3](https://docs.andes.app/en/latest/groupdoc/RenGen.html#regf3). For now, you can directly control the modules themself. New control algorithms might require additional development.

Failed initialization involves failed power flow and failed TDS initialization. Failed power flow is usually caused by a large gap in power balance. For this, you can check the power flow devices. Some additional discussions are available in [pandapower interface discussion](https://github.com/CURENT/andes/discussions/265#discussioncomment-4959298) for more information. As for failed TDS initialization, the following discussions might be helpful: [Question about Initialization](https://github.com/CURENT/andes/discussions/419), [EI System Initialization](https://github.com/CURENT/andes/discussions/407), [Changing the initialization point for TDS](https://github.com/CURENT/andes/discussions/402).

I summarized a blog of related questions in a blog [Fix Power Flow Divergence and TDS Initialization Error in ANDES](https://jinningwang.github.io/posts/2022/11/andes_init_debug/). Feel free to check it.

Hope you find these tips helpful.

Regards,
Jinning

**Reply by xiaoheimai95** [reply]
Thanks Jinning! And is it means that there is no need to configure exciter and governor for these GFM-controlled generating sets?

**Reply by jinningwang** [reply]
To my understanding, it is not mandatory right now. You can still directly control GFM by alternating setting points.

Regards,
Jinning

**Reply by cuihantao** [reply]
There is no need to configure exciters or governors fo GFM inverters. They have control loops for voltages and phase angles.

Initialization failure and convergence failure are two different issues. Initialization issues are commonly due to improper parameters (wrong buses, incorrect power rating, or limits that are too narrow).

Convergence failure is usually due to not having a solution for the power flow. It is commonly due to the disturbance being too significant. In this case, when a fault happens, the voltage is too long to transfer power. You can troubleshoot the case by increaseing the fault impedance, find an edge case where the system just won't converge, and identify parameters in the GFM converters to fix it.

**Reply by xiaoheimai95** [reply]
Thanks for answering!

**Reply by amparocorvalan** [reply]
Hi @cuihantao and @jinningwang,

I have further questions about the implementation of grid forming converters in ANDES. I would be very grateful if you can help me understand it better.

What I want is to evaluate an operating point of my system, where I know how much is the demand and generation in all busbars. If for this operating point I know that my battery injects a certain amount of power, can I model this power injection as a controlled PV generation with the GFM module provided by the library. Do you think this could be valid?

My goal is to see how the dynamic response of the system to disturbances improves when using grid forming converters and reactive compensation elements. 

Hope to get your advice!

Regards,
Amparo
